Sources/web3swift/Web3/Web3+Structures.swift

```diff
@@ -689,3 +689,58 @@ public struct TxPoolContentForNonce {
     public var nonce: BigUInt
     public var details: [TransactionDetails]
 }
+
+public enum FilterChanges: Decodable {
+    case hashes([String])
+    case logs([EventLog])
+    
+    public init(from decoder: Decoder) throws {
+        let container = try decoder.singleValueContainer()
+        if let hashes = try? container.decode([String].self) {
+            self = .hashes(hashes)
+        } else {
+            self = .logs(try container.decode([EventLog].self))
+        }
+    }
+}
+
+public struct BlockHeader: Decodable {
+    public let difficulty: String
+    public let extraData: String
+    public let gasLimit: String
+    public let gasUsed: String
+    public let logsBloom: String
+    public let miner: String
+    public let nonce: String
+    public let number: String
+    public let parentHash: String
+    public let receiptsRoot: String
+    public let sha3Uncles: String
+    public let stateRoot: String
+    public let timestamp: String
+    public let transactionsRoot: String
+}
+
+public struct LogItem: Decodable {
```

Comment:
This struct duplicates `EventLog` at [the line 331](https://github.com/skywinder/web3swift/blob/69b0a3f43c9fc2b087ba44f33f70a63745c50690/Sources/web3swift/Web3/Web3%2BStructures.swift#L331).

I've tested this branch and was able to get the same result using `EventLog` instead of `LogItem`. 

For the sake of interoperability I think it will be better to use `EventLog` for `web3.web3contract.getIndexedEvents`, `web3.web3contract.getIndexedEventsPromise` and for the new subscription API. Especially considering that these structs do not have `public init` (e.g. t won't be convenient to create `EventLog` out of `LogItem`) to avoid duplicates on the side of developers using this library it will make more sense to use `EventLog` as a single model for logs whether it's from web socket or not.

Reply:
Example input (web socket message received) and output (decoded):
```
received text: {
   "jsonrpc":"2.0",
   "method":"eth_subscription",
   "params":{
      "result":{
         "address":"0x6eb84ef768ffe65bca008af0dea3175eaf4df2ed",
         "blockHash":"0x6474699c14877b8b614c9ed417295913b283158a29964943f90c1802e6d1320a",
         "blockNumber":"0xbcf944",
         "data":"0x0000000000000000000000000000000000000000000000000000000000000020000000000000000000000000000000000000000000000000000000000000014414a6e2930000000000000000000000000000000000000000000000000000000000000040000000000000000000000000000000000000000000000000000000000000008000000000000000000000000000000000000000000000000000000000000000015ef83ad9559033e6e941db7d7c495acdce616347d28e90c7ce47cbfcfcad3bc50000000000000000000000000000000000000000000000000000000000000001000000000000000000000000000000000000000000000000000000000000002000000000000000000000000000000000000000000000000000000000000000596f357c6a7b0ce6b9327b83166db09820ec6703e1a1ff0c769747199f59130abc9c314ad5697066733a2f2f516d51474369465539657258756175396752537234695a48766b6a6e57636576625968784d324462664e47486f550000000000000000000000000000000000000000000000000000000000000000000000",
         "logIndex":"0x1",
         "removed":false,
         "topics":[
            "0x2e733a17851169f232b3859260eb3ad2a086afd54e999eb4ea9afb7791702e41",
            "0x0000000000000000000000000000000000000000000000000000000000000000"
         ],
         "transactionHash":"0x12b9f9312bfac44b95c82cf139a4a102efbe784a38e4eb973d1397a8b0f34a0a",
         "transactionIndex":"0x0",
         "transactionLogIndex":"0x1",
         "type":"mined"
      },
      "subscription":"0x451a9281a0d8a6a4"
   }
}
```

```
EventLog(address: web3swift.EthereumAddress(_address: "0x6eb84ef768ffe65bca008af0dea3175eaf4df2ed", type: web3swift.EthereumAddress.AddressType.normal), blockHash: 32 bytes, blockNumber: 12384580, data: 416 bytes, logIndex: 1, removed: false, topics: [32 bytes, 32 bytes], transactionHash: 32 bytes, transactionIndex: 0)
```

Reply:
Resolved in https://github.com/skywinder/web3swift/pull/446/commits/2d90894205261f4494e32cb66a88a78b66e5cf03
Thanks @odanylovych !